homeassistant/components/sensor/dht.py

```diff
@@ -18,18 +23,29 @@
                 '#Adafruit_DHT==1.3.0']
 
 _LOGGER = logging.getLogger(__name__)
+
 CONF_PIN = 'pin'
 CONF_SENSOR = 'sensor'
+
+DEFAULT_NAME = 'DHT Sensor'
+
+# DHT11 is able to deliver data once per second, DHT22 once every two
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)
+
 SENSOR_TEMPERATURE = 'temperature'
 SENSOR_HUMIDITY = 'humidity'
 SENSOR_TYPES = {
     SENSOR_TEMPERATURE: ['Temperature', None],
     SENSOR_HUMIDITY: ['Humidity', '%']
 }
-DEFAULT_NAME = "DHT Sensor"
-# Return cached results if last scan was less then this time ago
-# DHT11 is able to deliver data once per second, DHT22 once every two
-MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SENSOR): cv.string,
+    vol.Required(CONF_PIN): cv.string,
```

Comment:
Actually this needs to be a string as the pin numbering differs on various platforms.